hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java

```diff
@@ -628,7 +628,7 @@ public void run() {
                 new RecoveryTaskContiguous(b).recover();
               }
             } catch (IOException e) {
-              LOG.warn("recover Block: {} FAILED: {}", b, e);
+              LOG.warn("recover Block: {} FAILED: ", b, e);
```

Comment:
Whats wrong here? the number of placeholders are correct only, for the second one it will invoke e.toString(), now you are changing it to print the entire trace.

I don't think it is broken, it looks like it was intentional